### PR TITLE
Implemented replaceAll for Text

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Added `replaceAll` so we can use `Text` instead of `String` in `DateTime`.
@marcusnaslund ready for review
@tomasmahlberg can you also check that I got the `ownership` part right ?

### CODE REVIEW

File: source/base/Text.ooc [hunk 1]
```diff
@@ -149,6 +149,30 @@ Text: cover {
...
+		t := this take()
+		result: This
+		position := t find(toReplace)
+		if (position == -1)
+			result = t copy()
+		else {
+			previous := 0
+			result = This new()
+			x := toReplace take()
+			y := newValue take()
```
Comment (sebastianbaginski) [inline source/base/Text.ooc]: I saw that variables with one letter name are used in this file in other methods.

File: source/base/Text.ooc [hunk 2]
```diff
@@ -149,6 +149,30 @@ Text: cover {
...
+			while (position > -1) {
+				result += t slice(previous, position - previous) + y
```
Comment (marcusnaslund) [inline source/base/Text.ooc]: We have a `TextBuilder`, you know, wouldn't that be more efficient?

File: source/base/Text.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/base/Text.ooc]: Yeah, I can accept `t`, but `x` and `y` are pretty non-explanatory. `toReplaceTaken` and `newValueTaken`, is that too much?

File: test/base/TextTest.ooc [hunk 3]
```diff
@@ -177,17 +192,32 @@ this add("text literal toString", func {
...
 		})
+		this add("replace all", func {
+			text := t"aa00aa00"
+			notReplaced := text replaceAll(t"xx", t"yy")
+			expect(text == notReplaced)
+			replaced := text replaceAll(t"00", t"bb")
+			expect(replaced == t"aabbaabb")
+			replaced = t"xYxYxY" replaceAll(t"x", t"y")
+			expect(replaced == t"yYyYyY")
+			replaced = t"ooc is great language" replaceAll(t"great", t"the best") replaceAll(t"ooc", t"C++")
```
Comment (marcusnaslund) [inline test/base/TextTest.ooc]: Haha :)

File: source/base/Text.ooc (hunk 2 above)
Comment (sebastianbaginski) [inline source/base/Text.ooc]: I thought about it too, but would the code be more readable ? It's not that we will run `replaceAll` million times per frame

File: test/base/TextTest.ooc (hunk 3 above)
Comment (sebastianbaginski) [inline test/base/TextTest.ooc]: What ? It's just a random test string.

Comment (tomasmahlberg) [inline test/base/TextTest.ooc]: I LOL'd. :)

**Comment by tomasmahlberg** [comment]
Ownership looks about right in `replaceAll`. Maybe we can come up with a standard for naming taken variables in situations like this? Like adding a short suffix, maybe `newValue_` for `newValue` or something similar? Don't really like the look of `newValueTaken`.

File: source/base/Text.ooc (hunk 2 above)
Comment (marcusnaslund) [inline source/base/Text.ooc]: You're right.

**Comment by marcusnaslund** [comment]
@tomasmahlberg - will you make a cleanup issue about that? We should discuss `t` while we're at it.

**Comment by tomasmahlberg** [comment]
Issue #922.

**Comment by marcusnaslund** [comment]
Thanks.
I hope you'll be implementing this in `DateTime` (and `TimeSpan`) soon @sebastianbaginski :)

**Comment by sebastianbaginski** [comment]
Yes, I have next small PR stacked.
I also think we should start working on replacing `String` with `Text` in `File`. Memory leaks from the `File` class are quite annoying.

**Pull request merged**